Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -291,6 +291,307 @@ public final class SwiftTargetBuildDescription {
+                    write(record: TestEventRecord(bundleEvent: record))
+                }
+
+                public func testSuiteWillStart(_ testSuite: XCTestSuite) {
+                    let record = TestSuiteEventRecord(suite: .init(testSuite), event: .start)
+                    write(record: TestEventRecord(suiteEvent: record))
+                }
+
+                public func testCaseWillStart(_ testCase: XCTestCase) {
+                    let record = TestCaseEventRecord(testCase: .init(testCase), event: .start)
+                    write(record: TestEventRecord(caseEvent: record))
+                }
+
+                public func testCase(_ testCase: XCTestCase, didRecord issue: XCTIssue) {
+                    let record = TestCaseFailureRecord(testCase: .init(testCase), issue: .init(issue), failureKind: .unexpected)
+                    write(record: TestEventRecord(caseFailure: record))
+                }
+
+                public func testCase(_ testCase: XCTestCase, didRecord expectedFailure: XCTExpectedFailure) {
+                    let record = TestCaseFailureRecord(testCase: .init(testCase), issue: .init(expectedFailure.issue), failureKind: .expected(failureReason: expectedFailure.failureReason))
+                    write(record: TestEventRecord(caseFailure: record))
+                }
+
+                public func testCaseDidFinish(_ testCase: XCTestCase) {
+                    let record = TestCaseEventRecord(testCase: .init(testCase), event: .finish)
+                    write(record: TestEventRecord(caseEvent: record))
+                }
+
+                public func testSuite(_ testSuite: XCTestSuite, didRecord issue: XCTIssue) {
+                    let record = TestSuiteFailureRecord(suite: .init(testSuite), issue: .init(issue), failureKind: .unexpected)
+                    write(record: TestEventRecord(suiteFailure: record))
+                }
+
+                public func testSuite(_ testSuite: XCTestSuite, didRecord expectedFailure: XCTExpectedFailure) {
+                    let record = TestSuiteFailureRecord(suite: .init(testSuite), issue: .init(expectedFailure.issue), failureKind: .expected(failureReason: expectedFailure.failureReason))
+                    write(record: TestEventRecord(suiteFailure: record))
+                }
+
+                public func testSuiteDidFinish(_ testSuite: XCTestSuite) {
+                    let record = TestSuiteEventRecord(suite: .init(testSuite), event: .finish)
+                    write(record: TestEventRecord(suiteEvent: record))
+                }
+
+                public func testBundleDidFinish(_ testBundle: Bundle) {
+                    let record = TestBundleEventRecord(bundle: .init(testBundle), event: .finish)
+                    write(record: TestEventRecord(bundleEvent: record))
+                }
+            }
+
+            // FIXME: Copied from `XCTEvents.swift`, would be nice if we had a better way
```

Comment:
This is unfortunate, but not really a problem in practice most likely since the XCTest API doesn't change that frequently.

Reply:
cc @stmontgomery - is there?

Reply:
Oh maybe the comment is misleading, this is not an issue with XCTest API, but just the fact that we need this code in both `swift test` and the test bundle, but we don't have a good way to share it. I was just saying that it might not be an issue in practice since we don't have to change this too much. `XCTEvents.swift` is a file in this PR and it is basically the serialization I am using for events produced by XCTest.

Reply:
symlinks 🙄

Reply:
Don't think those can help, we need the source file at the time we're building the tests of a package, so we won't have any of SwiftPM's source code accessible. I guess in theory we could ship a library with SwiftPM which we link into any test bundle.

Reply:
Shipping a library would have impact if we ever supported "cross-testing", though, but maybe fine as long tests are always local because then we only need the library for the host platform.